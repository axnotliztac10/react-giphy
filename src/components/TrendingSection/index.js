import React from 'react';

import GiphyCard from 'components/GiphyCard';
import { Text, Box, Flex } from 'rebass';

export default class TrendingSection extends React.PureComponent {
  renderTrending = (trending, index) => {
    return (
      <GiphyCard
        giphy={trending}
        key= {index}
      />
    );
  }

  render() {
    const { trendings } = this.props;

    return (
      <Box my={4}>
        <Text textAlign="center" fontSize={25}>Trendings</Text>
        <Flex
          flexDirection="row"
          justifyContent="center"
          alignItems="flex-start"
          flexWrap="wrap"
        >
          {trendings.map(this.renderTrending)}
        </Flex>
      </Box>
    );
  }
}
