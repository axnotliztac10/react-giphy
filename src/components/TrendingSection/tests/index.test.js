import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import TrendingSection from '../index';

describe('<TrendingSection />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <TrendingSection
        trendings={[]}
        onAddRemoveFavorite={() => {}}
      />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
