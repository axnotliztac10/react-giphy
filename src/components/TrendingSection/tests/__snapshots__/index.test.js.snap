// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrendingSection /> should render correctly 1`] = `
<styled.div
  my={4}
>
  <Styled(styled.div)
    fontSize={25}
    textAlign="center"
  >
    Trendings
  </Styled(styled.div)>
  <Styled(styled.div)
    alignItems="flex-start"
    flexDirection="row"
    flexWrap="wrap"
    justifyContent="center"
  />
</styled.div>
`;
